Apromore-Custom-Plugins/CSVExporter-Logic/src/main/java/org/apromore/service/csvexporter/impl/CSVExporterLogicImpl.java

```diff
@@ -26,135 +26,225 @@
+            //empty StringBuilder
+            sb.setLength(0);
+
+            // write rest columns
+            String columnValue;
+            HashMap<String, String> attributeList;
+            HashMap<String, String> eventAttributes;
+            String attributeValue;
+
+            for (XTrace myTrace : xLog) {
+
+                attributeList = new HashMap<String, String>();
+
+                for (Map.Entry<String, XAttribute> tAtt : myTrace.getAttributes().entrySet()) {
+
+                    attributeValue = getAttributeValue(tAtt.getValue());
+                    if ("concept:name".equals(tAtt.getKey())) {
+                        attributeList.put(CASEID, attributeValue);
+                    } else {
+                        attributeList.put(tAtt.getKey(), attributeValue);
+                    }
+                }
+
+                for (XEvent myEvent : myTrace) {
+                    eventAttributes = new HashMap<String, String>(attributeList);
+
+                    for (Map.Entry<String, XAttribute> eAtt : myEvent.getAttributes().entrySet()) {
+
+                        attributeValue = getAttributeValue(eAtt.getValue());
+                        if ("concept:name".equals(eAtt.getKey())) {
+                            eventAttributes.put(ACTIVITY, attributeValue);
+                        } else {
+                            eventAttributes.put(eAtt.getKey(), attributeValue);
+                        }
+                    }
+
+                    LogModel row = new LogModel(eventAttributes);
+
+                    // start to write row
+                    prefix = "";
+                    for (String one : columnNames) {
+
+                        sb.append(prefix);
+                        prefix = ",";
+
+                        columnValue = row.getAttributeList().get(one);
+                        if (columnValue != null && columnValue.trim().length() != 0) {
+                            sb.append(columnValue);
+                        }
+                    }
+                    sb.append('\n');
+                    Files.write(tempCSV, sb.toString().getBytes(StandardCharsets.UTF_8),
+                            StandardOpenOption.APPEND);
+                    //empty StringBuilder
+                    sb.setLength(0);
 
-                columnValue = row.getAttributeList().get(one);
-                if(columnValue != null && columnValue.trim().length() !=0){
-                    sb.append(columnValue);
-                }else{
-                    sb.append("");
                 }
             }
-            sb.append('\n');
+
+            if (Files.notExists(tempCSV)) {
+                LOGGER.debug("The temp CSV path \"{}\" doesn't exist!", tempCSV);
+                return null;
+            }
+
+            compressGzip(tempCSV, tempGZIP);
+            Files.delete(tempCSV);
+            return tempGZIP.toFile();
+
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Do we still need to print the stack trace to console here if we're using the LOGGER to log the error?